components/calendar/src/chinese_based.rs

```diff
@@ -362,12 +494,20 @@ impl<C: ChineseBased + Calendar> CalendarArithmetic for C {
     }
 
     fn days_in_provided_year(year: i32) -> u16 {
-        let months_in_year = Self::months_for_every_year(year);
-        let mut days: u16 = 0;
-        for month in 1..=months_in_year {
-            days += Self::month_days(year, month) as u16;
-        }
```

Comment:
Question: why did you change the algorithm of this function?

Reply:
The newer version should be faster, but this isn't relevant anymore since I changed the function.